custom_components/localtuya/config_flow.py

```diff
@@ -274,36 +278,36 @@ async def async_step_add_entity(self, user_input=None):
 
     async def async_step_import(self, user_input):
         """Handle import from YAML."""
-
         def _convert_entity(conf):
             converted = {
                 CONF_ID: conf[CONF_ID],
                 CONF_FRIENDLY_NAME: conf[CONF_FRIENDLY_NAME],
-                CONF_PLATFORM: self.platform,
+                CONF_PLATFORM: conf[CONF_PLATFORM],
             }
-            for field in flow_schema(self.platform, self.dps_strings).keys():
-                converted[str(field)] = conf[field]
+#            print('CONVERTING [{}]'.format(conf[CONF_PLATFORM]))
+#            for field in flow_schema(conf[CONF_PLATFORM], self.dps_strings).keys():
+#                print('OPTIONS: [{}] [{}] '.format(flow_schema(conf[CONF_PLATFORM], self.dps_strings),conf.get(field)))
+#                converted[str(field)] = conf[field]
             return converted
 
         await self.async_set_unique_id(user_input[CONF_DEVICE_ID])
-        self.platform = user_input[CONF_PLATFORM]
-
-        if len(user_input.get(CONF_SWITCHES, [])) > 0:
-            for switch_conf in user_input[CONF_SWITCHES].values():
-                self.entities.append(_convert_entity(switch_conf))
-        else:
-            self.entities.append(_convert_entity(user_input))
 
         # print('ENTITIES: [{}] '.format(self.entities))
+        for entity_conf in user_input[CONF_ENTITIES]:
+#            self.platform = user_input[CONF_PLATFORM]
+            print('FOUND: [{}]'.format(entity_conf))
+            self.entities.append(_convert_entity(entity_conf))
```

Comment:
Since the entities in YAML maps directly to the format in the config entry, I guess we don't need any conversion at all? We can more or less pass `user_data` to `data` when creating the config entry below. The only bastard here is `CONF_YAML_IMPORT`, which I'm gonna remove soon (but we still have to keep). But just reply adding `CONF_YAML_IMPORT` with its value to `user_input`, remove everything else and pass `user_info` to `async_create_entry` and let's see what happens.

Reply:
Here is where the default values are not loaded for Options, I'd say.